docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,384 @@
+`childrenScheduleRoundMap`'s values equal to `scheduleCycle`, Gang's `scheduleCycle` will be added by 1, which means a new schedule cycle.
+
+We continue to explain `scheduleCycleValid` field, during the scheduling,  When a pod failed at Filter stage, we will set ScheduleCycleValid to 
+false in PostFilter stage, which means any pod in this Gang shouldn't be scheduled until it is set to "true",
+and the remaining pods should be rejected in PreFilter stage. Only When `scheduleCycle` added by 1, we will reset the `scheduleCycleValid` to true.
+
+It should be emphasized that `scheduleCycle\scheduleCycleValid\childrenScheduleRoundMap` only work in `strict-mode`. 
+
+##### GangPlugin
+
+this is the framework of the Plugin,we cache the Gang info above in the gangCache.
+```go
+type GangPlugin struct {
+    frameworkHandler            framework.Handle
+    gangClient                  gangClient.Interface
+    podLister                   listerv1.PodLister
+    snapshotSharedLister        framework.SharedLister
+    gangCache                   map[string]*Gang
+}
+```
+during the whole kubernetes shceduling process,we only need to realize our logic into four extention points as below:
+```go
+var(
+	_ framework.PreFilterPlugin = &GangScheduling{}
+	_ framework.PostFilterPlugin = &GangScheduling{}
+	_ framework.PermitPlugin = &GangScheduling{}
+	_ framework.ReservePlugin = &Coscheduling{}
+)
+
+type GangScheduling interface{
+    ActiveGang(pod *corev1.Pod, state *framework.CycleState)
+    PreFilter(context.Context, *corev1.Pod) error
+    PostFilter(ctx context.Context, state *CycleState, pod *v1.Pod, filteredNodeStatusMap NodeToStatusMap) (*PostFilterResult, *Status)
+    Permit(context.Context, *corev1.Pod) Status
+    Unreserve(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string)
+}
+
+```
+###### **PreFilter**
+
+if `non-strict-mode`, we only do step1 and step2:
+
+- Check whether childes in Gang has met the requirements of minimum number under each Gang, and reject the pod if negative.
+
+- Check whether the Gang has been timeout(check the pod's annotation,later introduced at Permit section), and reject the pod if positive.
+
+- Check whether the Gang has met the `scheduleCycleValid` check, and reject the pod if negative.
+
+- Try update `scheduleCycle`, `scheduleCycleValid`, `childrenScheduleRoundMap` as mentioned above.
+
+
+###### **PostFilter**
+
+At this point means the pod didn't pass the Filter Plugin, we should:
+
+- If `strict-mode`, we will set `scheduleCycleValid` to false and release all assumed pods.
+
+- If `non-strict mode`, we will do nothing.
+
+###### **Permit**
+
+Any pod passes Filter stage will come to this stage. Scheduler will calculate all Gangs in GangGroup whether the current 
+number of assumed-pods in each Gang meets the Gang's minimum requirement.
+
+- If Gang don't meet the bind-condition, we will give the pod a "Wait" Status with a timeout duration, and the bind 
```

Comment:
Please notice the maxTimeout is `15 minutes`, so we may need to hook `RunPermitPlugins` to enlarge the timeout when `15 minutes` is not enough.

Reply:
thanks! updated